pkg/operator/operatorclient/interfaces.go

```diff
@@ -76,6 +80,69 @@ func (c *CLIManagerClient) UpdateOperatorStatus(ctx context.Context, resourceVer
 	return &ret.Status.OperatorStatus, nil
 }
 
+func (c *CLIManagerClient) ApplyOperatorSpec(ctx context.Context, fieldManager string, applyConfiguration *applyoperatorv1.OperatorSpecApplyConfiguration) error {
+	original, err := c.OperatorClient.CliManagers(OperatorNamespace).Get(ctx, OperatorConfigName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	copied := original.DeepCopy()
```

Comment:
Not sure if you are expected to use an actual CliManager object. Compared to https://github.com/openshift/library-go/blob/756adf2188fcdb8f4af531514a76bc7bded706b7/pkg/operator/genericoperatorclient/dynamic_operator_client.go#L229-L271

Reply:
Based on https://github.com/openshift/client-go/blob/master/operator/applyconfigurations/operator/v1/operatorspec.go `OperatorSpecApplyConfiguration` has only 5 fields. Currently [CliManagerSpec](https://github.com/openshift/cli-manager-operator/blob/main/pkg/apis/climanager/v1/types_climanager.go) corresponds to [OperatorSpec](https://github.com/openshift/api/blob/master/operator/v1/types.go) which defines these 5 fields. Yet, if `CliManagerSpec` gets extended later, replacing the original spec will clear all the extra fields.

Reply:
Thanks. Your review has enlightened me. I think, I should first generate `CLIManagerApplyConfiguration` and should use it.

Reply:
I had to manually cast `OperatorSpecApplyConfiguration` to `CLIManagerApplyConfiguration` due to differences. Other than I that I believe that it is ready.